packages/nuxt/src/core/templates.ts

```diff
@@ -553,3 +554,11 @@ export const buildTypeTemplate: NuxtTemplate = {
     return declarations
   },
 }
+
+export const isolatedPageTemplate: NuxtTemplate = {
+  filename: 'isolated-page.mjs',
+  getContents (ctx) {
+    const hasPages = ctx.nuxt.options.dev || ctx.app.pages?.length
```

Comment:
I initially planned on using `ctx.app.pages?.some((page) => page.meta?.isolate))`   but the metas are not completely resolved at this stage. I'm checking if the app has some pages instead which should be fine since this means the vue-router integration is enabled and the NuxtPage component is already bundled.